core/src/main/java/org/apache/brooklyn/core/mgmt/entitlement/WebEntitlementContext.java

```diff
@@ -21,15 +21,24 @@
 import org.apache.brooklyn.api.mgmt.entitlement.EntitlementContext;
 import org.apache.brooklyn.util.javalang.JavaClassNames;
 
+import java.util.HashMap;
+import java.util.Map;
+
 /**
  * Indicates an authenticated web request as the entitlements context;
  * note user may still be null if no authentication was requested
  */
 public class WebEntitlementContext implements EntitlementContext {
 
+    public static final String ENTITLEMENTS_ATTRIBUTES = "brooklyn.entitlements.attributes";
+
+    public static final String USER_ROLES = "brooklyn.entitlements.user.roles";
+
+
     final String user;
     final String sourceIp;
     final String requestUri;
+    final Map<String, Object> attributes;
```

Comment:
probably `= MutableMap.of()` here

that will use `LinkedHashMap` which preserves order (i never use plain `HashMap`)

and then we can prevent a caller from adding map entries after they are passed in here